### PR TITLE
Model cleanup

### PR DESCRIPTION
**Pull request by tsherif**
First pass of cleanup including:
- Consolidate Model and BaseModel
- Remove seer integration
- Remove uniform animations
- Remove ShaderCache

Also fixed tests and examples.

Perf changes are as follows:

#### Stress Test CPU Render Time

|Before |After|Improvement|
|------ |-----|-----------|
|43ms   |34ms |9ms (21%)  |

#### Core Bundle Size

|Before|After|Improvement|
|------|-----|-----------|
|304kB |297kB|7kB (2%)   |

### CODE REVIEW

File: modules/core/src/lib/model.js [hunk 1]
```diff
@@ -1,20 +1,71 @@
...
-import BaseModel from './base-model';
 import {log, isObjectEmpty, uid, assert} from '../utils';
 
+const LOG_DRAW_PRIORITY = 2;
+const LOG_DRAW_TIMEOUT = 10000;
+
 const ERR_MODEL_PARAMS = 'Model needs drawMode and vertexCount';
 
-export default class Model extends BaseModel {
+export default class Model {
```
[the remark this answers is not in the document]
Comment (tsherif) [reply modules/core/src/lib/model.js]: BaseModel isn't used anywhere, so this added unnecessary complexity. I believe it was meant to support some Scenegraph use cases, but that didn't pan out.

File: modules/core/src/lib/base-model.js [hunk 2]
```diff
@@ -1,494 +0,0 @@
...
-  // Timer Queries
-
-  _timerQueryStart() {
-    if (this.timerQueryEnabled === true) {
-      if (!this.timeElapsedQuery) {
-        this.timeElapsedQuery = new Query(this.gl);
-      }
-      if (this.lastQueryReturned) {
-        this.lastQueryReturned = false;
-        this.timeElapsedQuery.beginTimeElapsedQuery();
```
[the remark this answers is not in the document]
Comment (tsherif) [reply modules/core/src/lib/base-model.js]: Yes, in the animation loop. The tracking done here is actually set up incorrectly and would interfere with the frame timer (which we're using quite heavily now) if it were activated.

**Comment by tsherif** [comment]
@ibgreen we can definitely discuss before this lands, but I wanted to get started as this is my primary task for this release. I will say that with the significant performance wins, the case for dropping seer support has become pretty compelling for me.

**Comment by ibgreen** [comment]
> significant performance wins, the case for dropping seer support has become pretty compelling for me.

Performance is a non-issue. seer can be supported with zero performance impact when not debugging. It is a simple question of not updating seer when seer is not active, and only updating once per second. 

There is also an override system in seer that allows user to edit things on the seer side that does not need to be active.

**Pull request merged**